### PR TITLE
Convert plugin template to Circle CI 2.0

### PR DESCRIPTION
**Pull request by jdee**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Related to #10563. This updates the plugin template to use Circle CI 2.0, including test reporting.

### Description

Replaced circle.yml with .circleci/config.yml.

To enable test reporting, added `rspec_junit_formatter` as a dev dependency. RSpec now generates both doc-style output and a JUnit-style report.xml. The latter is used by CI for test reporting.

I played around with a number of options until I found all the right info in various places in the Circle docs to produce results like this:

screen shot 2017-10-15 at 9 21 35 am

https://circleci.com/docs/2.0/configuration-reference/#store_test_results

Linked from that page ("classic CircleCI test metadata directory layout"):

https://circleci.com/docs/1.0/test-metadata/
https://circleci.com/docs/1.0/test-metadata/#rspec

### CODE REVIEW

**Comment by KrauseFx** [comment]
Awesome, thanks for your contribution on this @jdee, awesome idea. Let's see what @DanToml says about it :)

**Comment by jdee** [comment]
One thought I had after submitting this: There's also Travis. That is, there's also a .travis.yml in the template. So devs have the option of not using what's here, and they may not use the `rspec_junit_formatter` gem. Of course, this is just a template, and they can modify whatever they like afterward. (They probably won't use both the .travis.yml and .circleci/config.yml, but they're both present, and one or both will probably be deleted.) It's nice for things to be set up well by default. And of course test reporting can also be set up for Travis. I think JUnit-style reporting is supported by most CI systems. I've used this gem with Jenkins before. So I don't think there's any harm including this dependency by default. But open to other opinions.

File: fastlane/lib/fastlane/plugins/template/.circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "Gemfile" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
+
+      - run:
+          name: install dependencies
+          command: |
+            bundle install --jobs=4 --retry=3 --path vendor/bundle
```
Comment (DanToml) [inline fastlane/lib/fastlane/plugins/template/.circleci/config.yml]: this should be `command: bundle check || bundle install --jobs=4 --retry=3 --path vendor/bundle`

File: fastlane/lib/fastlane/plugins/template/.circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+      - save_cache:
+          paths:
+            - ./venv
```
Comment (DanToml) [inline fastlane/lib/fastlane/plugins/template/.circleci/config.yml]: I think we want the vendor folder here

File: fastlane/lib/fastlane/plugins/template/.circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          key: v1-dependencies-{{ checksum "Gemfile" }}
+
+      # run tests!
+      - run:
+          name: run tests
+          command: |
```
Comment (DanToml) [inline fastlane/lib/fastlane/plugins/template/.circleci/config.yml]: no need for multi line command here either

**Review by endocrimes — CHANGES_REQUESTED** [review]
Good start!

**Comment by jdee** [comment]
@dantoml Thanks. I think I addressed your concerns, particularly `bundle check` before `bundle install` and caching the vendor folder. (I think I was working from an old Circle template and never noticed that. All my gems thank you for catching it.)

**Review by endocrimes — APPROVED** [review]

**Pull request merged**